packages/api-server/src/modules/assets.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  AssetGetRequest,
+  DendronError,
+  ERROR_STATUS,
+  RespV2,
+} from "@dendronhq/common-all";
+import { WorkspaceUtils } from "@dendronhq/engine-server";
+import { getWS } from "../utils";
+
+export class AssetsController {
+  static singleton?: AssetsController;
+
+  static instance() {
+    if (!AssetsController.singleton) {
+      AssetsController.singleton = new AssetsController();
+    }
+    return AssetsController.singleton;
+  }
+
+  async get({ fpath, ws }: AssetGetRequest): Promise<RespV2<string>> {
+    const engine = await getWS({ ws });
+    const { wsRoot, vaults } = engine;
+    if (!WorkspaceUtils.isPathInWorkspace({ wsRoot, vaults, fpath })) {
+      return {
+        error: DendronError.createFromStatus({
+          status: ERROR_STATUS.INVALID_CONFIG,
```

Comment:
Is `INVALID_CONFIG` the correct error here? Seems either DNE or something related to missing asset is more appropriate?

Reply:
this is protecting against bad access. one could you the `getAsset` api to retrieve the contents of any file on your system - this limits it to only paths underneath your workspace

Reply:
can someone call this API with an fpath that doesn't point to a valid location? That seems to be a possibility right?

Reply:
yep, that's a possibility as well. will add an additional `DNE` check

Reply:
see this [pr](https://github.com/dendronhq/dendron/pull/931) for changes